packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -174,6 +174,14 @@ export interface AutoScalingGroupProps extends CommonAutoScalingGroupProps {
    * @default A role will automatically be created, it can be accessed via the `role` property
    */
   readonly role?: iam.IRole;
+
+  /**
+   * The maximum hourly price to be paid for any Spot Instance launched to fulfill the request. Spot Instances are
+   * launched when the price you specify exceeds the current Spot market price.
+   *
+   * @default none
+   */
+  readonly spotPrice?: string;
```

Comment:
Wouldn't it have made sense to add this to `CommonAutoScalingGroupProps`?

Also, is this literally all you need to do to support spot instance fleets? What happens if the price exceeds the number you put here? Will there be no instances running? How does that interact with MinSize?

I would appreciate a couple more lines on the behavior here, or a link to the documentation.

Reply:
Also, in the interest of annotating numbers with their unit, I feel this should be called `spotPriceUsd` or something, no?

Reply:
Yes, I added an integ test to show that this in fact works.

Adding USD feels a little verbose, but if we do that elsewhere I can change it. And yes, can also move this to `CommonAutoScalingGroupProps`.